ui_test/.gitignore

```diff
@@ -0,0 +1 @@
+Cargo.lock
```

Comment:
This is a workspace crate, why would such a file exist?

Reply:
Shouldn't that use the workspace lockfile?

Or do we actually not have a workspace? cargo-miri is also just a separate crate, so I guess that is the case.
And using workspaces anyway means trouble because of our rustc integration...

But does this mean `./miri test` does currently not run these tests?

Reply:
it doesn't run those tests.

Reply:
We have the other lock files committed. This is required to ensure tests are reproducible. Why wouldn't we want to commit this one as well?

Reply:
for our main test suite it's irrelevant. This is just for unit tests of the ui_test crate. library crates usually don't commit their Cargo.lock

Reply:
> library crates usually don't commit their Cargo.lock

I know. I was about to say I disagree with that common choice since it means old CI runs cannot be (easily) reproduced, but we can defer this to a later PR. Then you added the lockfile already anyway. Thanks. :)